packages/context-editor/src/ContextEditor.tsx

```diff
@@ -82,103 +60,55 @@ const initSuggestProps: SuggestProps = {
+			{ children, nodeProps, ...rest },
+			ref
+		) {
 			const AtomRenderingComponent = props.atomRenderingComponent;
-			return <AtomRenderingComponent nodeProp={undefined} />;
-		};
+			return <AtomRenderingComponent ref={ref} nodeProps={nodeProps} {...rest} />;
+		});
 	}, [props.atomRenderingComponent]);
-	const nodeViewFactory = useNodeViewFactory();
-	const pluginViewFactory = usePluginViewFactory();
-	const viewHost = useRef<HTMLDivElement | null>(null);
-	const view = useRef<EditorView | null>(null);
-	const [panelPosition, setPanelPosition] = useState<PanelProps>(initPanelProps);
-	const [suggestData, setSuggestData] = useState<SuggestProps>(initSuggestProps);
 
-	const menuBarId = useId();
+	const [suggestData, setSuggestData] = useState<SuggestProps>(initSuggestProps);
 
-	/* This useEffect approach of making the props available to prosemirror  */
-	/* plugins from: https://discuss.prosemirror.net/t/lightweight-react-integration-example/2680 */
-	useEffect(() => {
-		/* Initial Render */
-		const state = EditorState.create({
+	const [editorState, setEditorState] = useState(
+		EditorState.create({
 			doc: props.initialDoc ? baseSchema.nodeFromJSON(props.initialDoc) : undefined,
 			schema: baseSchema,
-			plugins: [
-				...basePlugins(
-					baseSchema,
-					props,
-					panelPosition,
-					setPanelPosition,
-					suggestData,
-					setSuggestData
-				),
-				...(props.hideMenu
-					? []
-					: [
-							new Plugin({
-								view: pluginViewFactory({
-									component: () => <MenuBar upload={props.upload} />,
-									root: () => {
-										return document.getElementById(menuBarId) as HTMLElement;
-									},
-								}),
-							}),
-						]),
-			],
-		});
-		if (viewHost.current) {
-			view.current = new EditorView(viewHost.current, {
-				state,
-				nodeViews: {
-					contextAtom: nodeViewFactory({
-						component: Renderer,
-					}),
-				},
-				handleDOMEvents: {
-					focus: () => {
-						/* Reset the panelProps when the editor is focused */
-						setPanelPosition(initPanelProps);
-					},
-				},
-			});
-		}
-		// applyDevTools(view.current);
-		return () => view.current?.destroy();
-	}, []);
+			plugins: [...basePlugins(baseSchema, props, suggestData, setSuggestData), reactKeys()],
+		})
+	);
 
 	useEffect(() => {
-		/* Every Render */
-		if (view.current) {
-			view.current.setProps({
-				editable: () => !props.disabled,
-			});
-			const tr = view.current.state.tr
-				.setMeta(reactPropsKey, { ...props, suggestData, setSuggestData })
-				.setMeta(attributePanelKey, { panelPosition, setPanelPosition });
-			view.current?.dispatch(tr);
-		}
-		/* It's not clear to me that any of the props need to trigger this to re-render.  */
-		/* Doing so in some cases (onChange for the EditorDash) cause an infinite re-render loop */
-		/* Figure out what I actually need to render on, and then clean up any useMemo calls if necessary */
-	}, [props, suggestData, panelPosition]);
+		props.onChange(editorState);
```

Comment:
moved the `onChange` call to this `useEffect` rather than using the `onChange` plugin. this is possible now because our editor state is hoisted up.

I originally tried to put `onChange` inside of the `setEditorState` call on L95 but this resulted in a "bad set state" error and I'm not sure I totally know why